### ISSUE TITLE
Confusing "error: Alternative syntax for unions requires Python 3.10 or newer" in type aliases - should be different message.

### ISSUE THREAD

**Issue by laundmo** [microsoft/pyright#3822]
**Describe the bug**
A type alias like `Number = int | float` will throw the `error: Alternative syntax for unions requires Python 3.10 or newer` which is the same error as a missing `from __future__ import annotations` import leading to confusion. Technically, the error message is not wrong, but due to it being used for 2 separate causes of the error its confusing - especially since both union syntaxes work in 3.10 without a import.

**To Reproduce**
Use python 3.9
Make file with content below
Run pyright on file
observe wrong error message

**Expected behavior**
Error message which specifies that the new union syntax cant be used in type-aliases with the future import.

**Screenshots or Code**

```python
from __future__ import annotations

def f(a: int | float):
    pass

Number = int | float
```

```
pyright 1.1.266
test.py
  test.py:6:14 - error: Alternative syntax for unions requires Python 3.10 or newer
```

**VS Code extension or command-line**
issue exists in both:
CLI: pyright 1.1.257
Pylance language server 2022.8.21 (pyright ed13daf1)

**Comment by hmc-cs-mdrissi** [comment]
Try running file in 3.9 or lower. It'll crash on type alias. `from __future__ import annotations` only works on annotations, not on other usages of types like in aliases.

**Comment by laundmo** [comment]
yeah i noticed seconds after posting, very dumb mistake on my part

**Comment by laundmo** [comment]
Maybe a premature close, gonna re-work this into showing a clearer message in those cases

**Comment by erictraut** [comment]
I think the error message is accurate here.
